client/push/fcm.js

```diff
@@ -0,0 +1,30 @@
+(function (global, factory) {
+  'use strict';
+  global.fcm = new (factory(global, global.document))();
+})(typeof window !== 'undefined' ? window : this, function (w, d) {
+  var fcm = function () {
+    var self = this;
+
+    function getFirebaseToken () {
+    }
+
+    function deleteFirebaseToken () {
+    }
+
+    function requestPushPermission () {
```

Comment:
Please follow the coding rule.
There must be no spaces between function name and ().